File: leetcode/twoSum.go
package leetcode

func twoSum(nums []int, target int) []int {
	var value []int = nil
	var valueIndexMap = make(map[int]int)

	for index, num := range nums {

		var matchNum = target - num

		if matchNumIndex, ok := valueIndexMap[matchNum]; ok {
			value = []int{matchNumIndex, index}
			break
		}

		valueIndexMap[num] = index
	}

	return value
}

File: leetcode/0001-two-sum/twoSum.ts
function twoSum(nums: number[], target: number): number[] {
  const valueIndexMap = new Map<number, number>();
  let value: number[] = [];

  for (let index = 0; index < nums.length; index++) {
    const num = nums[index]!;
    const matchNum = target - num;

    const matchNumIndex = valueIndexMap.get(matchNum);

    if (matchNumIndex != undefined) {
      value = [matchNumIndex, index];
      break;
    }

    valueIndexMap.set(num, index);
  }

  return value;
}

export default twoSum;
